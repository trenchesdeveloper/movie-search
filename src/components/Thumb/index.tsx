import React from "react";
//styles
import { Image } from "./Thumb.styles";
//Types
type Props = {
  image: string;
};

const Thumb: React.FC<Props> = ({ image }) => {
  return (
    <div>
      <Image src={image} alt="movie-thumb" />
    </div>
  );
};

export default Thumb;
