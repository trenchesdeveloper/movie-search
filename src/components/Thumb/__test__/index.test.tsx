import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import Thumb from "..";

describe("Thumb", () => {
  test("Thumb component renders correctly", () => {
    render(<Thumb image="image" />);
    const image = screen.getByRole("img", { name: "movie-thumb" });

    expect(image).toHaveAttribute("src", "image");

    expect(image).toBeInTheDocument();

    expect(image).toHaveAttribute("alt", "movie-thumb");
  });

  test("Thumb component snapshot", () => {
    const tree = renderer.create(<Thumb image="" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
