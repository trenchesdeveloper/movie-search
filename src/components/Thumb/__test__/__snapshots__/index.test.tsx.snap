// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumb Thumb component snapshot 1`] = `
.c0 {
  width: 100%;
  max-width: 720px;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  object-fit: cover;
  border-radius: 20px;
  -webkit-animation: animateThumb 0.5s;
  animation: animateThumb 0.5s;
}

.c0:hover {
  opacity: 0.8;
}

<div>
  <img
    alt="movie-thumb"
    className="c0"
    src=""
  />
</div>
`;

exports[`Thumb component snapshot 1`] = `
.c0 {
  width: 100%;
  max-width: 720px;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  object-fit: cover;
  border-radius: 20px;
  -webkit-animation: animateThumb 0.5s;
  animation: animateThumb 0.5s;
}

.c0:hover {
  opacity: 0.8;
}

<div>
  <img
    alt="movie-thumb"
    className="c0"
    src=""
  />
</div>
`;
